dll/cpl/sysdm/environment.c

```diff
@@ -407,18 +849,41 @@ OnEditVariable(HWND hwndDlg,
 
         if (ListView_GetItem(hwndListView, &lvi))
         {
-            VarData = (PVARIABLE_DATA)lvi.lParam;
+            DlgData->VarData = (PVARIABLE_DATA)lvi.lParam;
 
-            if (DialogBoxParam(hApplet,
-                               MAKEINTRESOURCE(IDD_EDIT_VARIABLE),
-                               hwndDlg,
-                               EditVariableDlgProc,
-                               (LPARAM)VarData) > 0)
+            /* If the value has multiple values and directories then edit value with fancy dialog box */
+            if (wcschr(DlgData->VarData->lpRawValue, ';') != NULL && wcschr(DlgData->VarData->lpRawValue, '\\') != NULL)
```

Comment:
Your strings are TCHARs, so for consistency you would need to use _tcschr() functions here.